### ISSUE TITLE
[BUG] Migration Fails When Formbricks Is Not yet on Latest Version

### ISSUE THREAD

**Issue by jonas-hoebenreich** [formbricks/formbricks#2961]
### Issue Summary

The database migrations are currently separated into migrations that run within Formbricks and migrations that run within data-migrations. In order to run data-migrations you have to start the Formbricks container on the latest version first. But Formbricks does not work properly as long as data-migrations is not finished. This leads to issues when you try to run the migrations.

### Steps to Reproduce

1. Run data-migrations without running Formbricks on the latest version
2. Migration fails bc some migrations applied by Formbricks are missing

### Expected behavior

data-migrations runs all necessary migrations without the need to run Formbricks on the latest version first.

### Environment

- [ ] Formbricks Cloud (app.formbricks.com)
- [X] Self-hosted Formbricks

### Desktop (please complete the following information)

```markdown
- v2.4.0
```

**Comment by jonas-hoebenreich** [comment]
Yes, I know it's not a straightforward thing to fix. But having a mutual dependency where the migration depends on the newest formbricks container and vice versa is not very satisfactory ;)

**Comment by mattinannt** [comment]
We have shipped the new automatic approach with Formbricks 3.0.
Now all migrations (schema migrations as well as data migrations) run automatically on container start :-)
